tooling/repl.lisp

```diff
@@ -188,26 +188,32 @@
     (cons
      (destructuring-bind (head . rest)
          form
-     (case head
-       (:assert
-        (assert (not (eq (repl-nil repl) (eval-expr (car rest) state))))
-        state)
-       (:assert-eq
-        (assert (equal (eval-expr (first rest) state) (eval-expr (second rest) state)))
-        state)
-       (:assert-error
-        (assert (handler-case (prog1 nil (eval-expr (first rest) state))
-                  (error () t)))
-        state)
-       (:clear
-        (let ((new-state (copy-repl-state state)))
-          (setf (repl-state-env new-state) (api-impl:empty-env))
-          new-state))
-       (:load
-        (let ((new-state (load-lib state (car rest))))
-          new-state))
-       (t (format (repl-state-out state) "Unhandled: ~S" head)
-        state))))
+       (ecase head
```

Comment:
The change from `case` to `ecase` is what makes this more generally correct. Now unknown directives will lead to failure.